### DISCUSSION TITLE
Syntax highlighting broken with generics?

### DISCUSSION THREAD

**Discussion by xlucas** [golang/vscode-go#2114] [Help] [answered]
Hello 👋 

I was fiddling with generics and noticed function names and arguments are not highlighted on all themes:

![image](https://p-qkfgo2.t2.n0.cdn.getcloudapp.com/items/DOu4G8Am/e1814381-95d2-4a90-94cf-66c38c90927c.png?v=2af4f5ad74ca6c00e0849639f0706994)

I could not find an issue for that so wondering if it's actually impacting others?

**Comment by suzmue** [comment, marked as answer]
Hi @xlucas! 

It appears that the TextMate based syntax highlighting does not appropriately highlight generic code. If you use the semantic tokens provided by gopls instead, it does support generics!

Add the following to your settings:
```
    "gopls": {
        "ui.semanticTokens": true
    }
```

**Reply by hyangah** [reply]
For textmate based syntax highlighting, open an issue in https://github.com/jeff-hykin/better-go-syntax.

**Reply by xlucas** [reply]
Thanks! I also had to set `"editor.semanticHighlighting.enabled": true` and it's fine now 🙇
